File: typescript/lld/money-ledger/payment-gateway/RazorpayGateway.ts
import { PaymentGateway, ProviderResult } from './PaymentGateway';

// A second adapter — succeeds immediately, standing in as the failover target.
export class RazorpayGateway implements PaymentGateway {
    readonly name = 'Razorpay';

    charge(amountCents: number, idempotencyKey: string): ProviderResult {
        return { providerRef: `razorpay_${idempotencyKey}` };
    }
}

File: python/lld/money-ledger/payment-gateway/razorpay_gateway.py
from payment_gateway import PaymentGateway, ProviderResult


class RazorpayGateway(PaymentGateway):
    """A second adapter — succeeds immediately, standing in as the
    failover target."""

    name = "Razorpay"

    def charge(self, amount_cents: int, idempotency_key: str) -> ProviderResult:
        return ProviderResult(f"razorpay_{idempotency_key}")